mempool/src/shared_mempool.rs

```diff
@@ -36,10 +60,22 @@ use vm_validator::vm_validator::{get_account_state, TransactionValidation};
 struct PeerSyncState {
     timeline_id: u64,
     is_alive: bool,
+    network_id: PeerId,
 }
 
+/// stores only peers that receive txns from this node
 type PeerInfo = HashMap<PeerId, PeerSyncState>;
 
+/// Handle in a node's shared mempool to components of a network that the node belongs to.
+pub struct MempoolNetworkHandle {
+    /// sender for this network
+    pub network_sender: MempoolNetworkSender,
+    /// events from the network to this node
+    pub network_events: MempoolNetworkEvents,
+    /// true if this network is a validator network, else false
+    pub is_validator_network: bool,
```

Comment:
is this field still required?
also I'd argue you don't need this struct anymore
if you really want to, you can create alias to tuple of  (MempoolNetworkSender, MempoolNetworkEvents) but it's up to you